CHECKLIST.md

```diff
@@ -1611,5 +1611,38 @@ _Details coming soon…_
 
 _Details coming soon…_
 
+## 3. Method Design Checklist
+
+### 3.6.12. **Consider** a builder when faced with many constructor parameters
+
+#### Rationale:
+
+Telescoping constructors are hard to use to read and maintain. They are also error-prone, as it's easy to skip some
+combinations. The problem grows with the list of parameters, when we need to create a constructor for each combination.
+
+A builder can solve this problem.
+
+##### Builder
+
+Pros:
+ 
+ - no need for deprecation if we add a new required parameter
+ - fewer constructors to maintain
+ - less pain for the consumers, because they no longer have to pick between different constructor signatures
```

Comment:
Throughout the document the "consumers" are called either "clients" or "callers". Let's reuse one of the existing terms.